Startup fails in a Spring boot application.

I am trying to start Jet as a simple spring boot application with no other dependencies.

Added a Bean that creates a Jet instance with default configuration.

>  @Bean
    public JetInstance getJetInstance(){
        return Jet.newJetInstance();
    }

Jet version is 0.4

Below is the error:
Caused by: java.lang.IllegalStateException: Failed to introspect Class [com.hazelcast.jet.impl.JetInstanceImpl] from ClassLoader [sun.misc.Launcher$AppClassLoader@18b4aac2]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:656) ~[spring-core-5.0.0.RC2.jar:5.0.0.RC2]

Caused by: java.lang.ClassNotFoundException: javax.cache.Cache
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381) ~[na:1.8.0_74]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_74]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331) ~[na:1.8.0_74]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_74]
	... 47 common frames omitted

--- a/hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetBootstrap.java
+++ b/hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetBootstrap.java
@@ -25,7 +25,7 @@ import com.hazelcast.jet.Job;
 import com.hazelcast.jet.config.JetConfig;
 import com.hazelcast.jet.config.JobConfig;
 import com.hazelcast.jet.impl.util.Util;
-import com.hazelcast.jet.stream.IStreamCache;
+import com.hazelcast.jet.stream.JetCacheManager;
 import com.hazelcast.jet.stream.IStreamList;
 import com.hazelcast.jet.stream.IStreamMap;
 
@@ -189,8 +189,8 @@ public final class JetBootstrap {
         }
 
         @Override
-        public <K, V> IStreamCache<K, V> getCache(String name) {
-            return instance.getCache(name);
+        public JetCacheManager getCacheManager() {
+            return instance.getCacheManager();
         }
 
         @Override

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/JetInstance.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/JetInstance.java
@@ -20,7 +20,7 @@ import com.hazelcast.core.Cluster;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.jet.config.JetConfig;
 import com.hazelcast.jet.config.JobConfig;
-import com.hazelcast.jet.stream.IStreamCache;
+import com.hazelcast.jet.stream.JetCacheManager;
 import com.hazelcast.jet.stream.IStreamList;
 import com.hazelcast.jet.stream.IStreamMap;
 
@@ -78,14 +78,6 @@ public interface JetInstance {
     <K, V> IStreamMap<K, V> getMap(String name);
 
     /**
-     * Returns the distributed cache instance with the specified name.
-     *
-     * @param name name of the distributed cache
-     * @return distributed cache instance with the specified name
-     */
-    <K, V> IStreamCache<K, V> getCache(String name);
-
-    /**
      * Returns the distributed list instance with the specified name.
      * Index-based operations on the list are not supported.
      *
@@ -93,6 +85,17 @@ public interface JetInstance {
      * @return distributed list instance with the specified name
      */
     <E> IStreamList<E> getList(String name);
+
+    /**
+     * Obtain the {@link JetCacheManager} that provides access to JSR-107 (JCache) caches
+     * configured on a Hazelcast Jet cluster.
+     * <p>
+     * Note that this method does not return a JCache {@code CacheManager}
+     *
+     * @return the Hazelcast Jet {@link JetCacheManager}
+     * @see JetCacheManager
+     */
+    JetCacheManager getCacheManager();
 
     /**
      * Shutdowns the current instance.

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJetInstance.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJetInstance.java
@@ -20,10 +20,9 @@ import com.hazelcast.core.Cluster;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.core.IdGenerator;
 import com.hazelcast.jet.JetInstance;
-import com.hazelcast.jet.stream.IStreamCache;
+import com.hazelcast.jet.stream.JetCacheManager;
 import com.hazelcast.jet.stream.IStreamList;
 import com.hazelcast.jet.stream.IStreamMap;
-import com.hazelcast.jet.stream.impl.CacheDecorator;
 import com.hazelcast.jet.stream.impl.ListDecorator;
 import com.hazelcast.jet.stream.impl.MapDecorator;
 
@@ -32,8 +31,11 @@ abstract class AbstractJetInstance implements JetInstance {
 
     private final HazelcastInstance hazelcastInstance;
 
+    private final JetCacheManagerImpl cacheManager;
+
     AbstractJetInstance(HazelcastInstance hazelcastInstance) {
         this.hazelcastInstance = hazelcastInstance;
+        this.cacheManager = new JetCacheManagerImpl(this);
     }
 
     @Override
@@ -57,13 +59,13 @@ abstract class AbstractJetInstance implements JetInstance {
     }
 
     @Override
-    public <K, V> IStreamCache<K, V> getCache(String name) {
-        return CacheGetter.getCache(this, name);
+    public <E> IStreamList<E> getList(String name) {
+        return new ListDecorator<>(hazelcastInstance.getList(name), this);
     }
 
     @Override
-    public <E> IStreamList<E> getList(String name) {
-        return new ListDecorator<>(hazelcastInstance.getList(name), this);
+    public JetCacheManager getCacheManager() {
+        return cacheManager;
     }
 
     @Override
@@ -73,15 +75,5 @@ abstract class AbstractJetInstance implements JetInstance {
 
     protected IdGenerator getIdGenerator() {
         return hazelcastInstance.getIdGenerator(JET_ID_GENERATOR_NAME);
-    }
-
-    /**
-     * This class is necessary to conceal the cache-api
-     */
-    private static class CacheGetter {
-
-        private static <K, V> IStreamCache<K, V> getCache(JetInstance instance, String name) {
-            return new CacheDecorator<>(instance.getHazelcastInstance().getCacheManager().getCache(name), instance);
-        }
     }
 }

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetCacheManagerImpl.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetCacheManagerImpl.java
@@ -1,0 +1,53 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl;
+
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.stream.IStreamCache;
+import com.hazelcast.jet.stream.JetCacheManager;
+import com.hazelcast.jet.stream.impl.CacheDecorator;
+
+/**
+ * Hazelcast Jet {@code JetCacheManager} implementation accessible via {@code JetInstance} that provides access to JSR-107
+ * (JCache) caches configured on the cluster via Hazelcast API.
+ *
+ * @see JetCacheManager
+ */
+public class JetCacheManagerImpl implements JetCacheManager {
+
+    private final JetInstance jetInstance;
+
+    JetCacheManagerImpl(JetInstance jetInstance) {
+        this.jetInstance = jetInstance;
+    }
+
+    @Override
+    public <K, V> IStreamCache<K, V> getCache(String name) {
+        return CacheGetter.getCache(jetInstance, name);
+    }
+
+    /**
+     * This class is necessary to conceal the cache-api
+     */
+    private static class CacheGetter {
+
+        private static <K, V> IStreamCache<K, V> getCache(JetInstance jetInstance, String name) {
+            return new CacheDecorator<>(jetInstance.getHazelcastInstance().getCacheManager().getCache(name), jetInstance);
+        }
+
+    }
+}

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/stream/DistributedCollectors.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/stream/DistributedCollectors.java
@@ -1388,7 +1388,7 @@ public abstract class DistributedCollectors {
     private static class CacheGetter {
 
         private static <K, V> DistributedFunction<JetInstance, IStreamCache<K, V>> getCacheF(String name) {
-            return instance -> instance.getCache(name);
+            return instance -> instance.getCacheManager().getCache(name);
         }
     }
 }

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/stream/JetCacheManager.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/stream/JetCacheManager.java
@@ -1,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.stream;
+
+
+import com.hazelcast.jet.JetInstance;
+
+/**
+ * {@code JetCacheManager} is the entry point to access JSR-107 (JCache) caches via {@link JetInstance} interface.
+ * Hazelcast Jet's {@code JetCacheManager} provides access to JCache caches configured cluster-wide,
+ * even when created by different JCache {@link javax.cache.CacheManager}s.
+ * <p>
+ * <p>Note that this interface is not related to JCache {@link javax.cache.CacheManager}. Its purpose is to host
+ * {@code IStreamCache} related methods, separately from {@link JetInstance}, in order to allow frameworks that make
+ * use of reflection and/or dynamic proxies (e.g. Mockito, Spring etc) to operate on {@link JetInstance} when JCache
+ * is not on the classpath.
+ * </p>
+ *
+ * @since 0.5
+ */
+public interface JetCacheManager {
+
+    /**
+     * <p>
+     * Returns the cache instance with the specified prefixed cache name.
+     * </p>
+     *
+     * <p>
+     * Prefixed cache name is the name with URI and classloader prefixes if available.
+     * There is no Hazelcast prefix ({@code /hz/}). For example, {@code myURI/foo}.
+     *
+     * <pre>
+     * <code>
+     *     &lt;prefixed_cache_name&gt; = [&lt;uri_prefix&gt;/] + [&lt;cl_prefix&gt;/] + &lt;pure_cache_name&gt;
+     * </code>
+     * </pre>
+     * where {@code <pure_cache_name>} is the cache name without any prefix. For example {@code foo}.
+     *
+     * As seen from the definition, URI and classloader prefixes are optional.
+     *
+     * URI prefix is generated as content of this URI as a US-ASCII string. ({@code uri.toASCIIString()})
+     * Classloader prefix is generated as string representation of the specified classloader. ({@code cl.toString()})
+     *
+     * @see com.hazelcast.cache.CacheUtil#getPrefixedCacheName(String, java.net.URI, ClassLoader)
+     * </p>
+     *
+     * @param name the prefixed name of the cache
+     * @return the cache instance with the specified prefixed name
+     *
+     * @throws com.hazelcast.cache.CacheNotExistsException  if there is no configured or created cache
+     *                                                      with the specified prefixed name
+     * @throws java.lang.IllegalStateException              if a valid JCache library does not exist in the classpath
+     *                                                      ({@code 1.0.0-PFD} or {@code 0.x} versions are not valid)
+     */
+    <K, V> IStreamCache<K, V> getCache(String name);
+}

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/JetTestSupport.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/JetTestSupport.java
@@ -69,7 +69,7 @@ public class JetTestSupport extends HazelcastTestSupport {
     }
 
     protected static <K, V> IStreamCache<K, V> getCache(JetInstance instance) {
-        return instance.getCache(randomName());
+        return instance.getCacheManager().getCache(randomName());
     }
 
 

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/stream/AbstractStreamTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/stream/AbstractStreamTest.java
@@ -119,7 +119,7 @@ public abstract class AbstractStreamTest extends JetTestSupport {
         String cacheName = randomName();
         IStreamCache<K, V> cache = null;
         for (JetInstance jetInstance : instances) {
-            cache = jetInstance.getCache(cacheName);
+            cache = jetInstance.getCacheManager().getCache(cacheName);
         }
         return cache;
     }

